src/var/binary_snps.jl

```diff
@@ -0,0 +1,167 @@
+
+#=
+Objects of type BinarySNPs store the SNPs of an individual
+The name BinarySNPs is double edged: it refers to both the fact that
+SNPs are stored in a binary format, and also that the SNPs are binary,
+in that they have two states, a reference state and an alternate state.
```

Comment:
This type stores the snaps of a single individual, rather than a matrix of individuals. SNPs are stored in binary format using vectors. So 0 indicates the position has the reference allele, whilst a 1 indicates the position has the alternative allele. If the individual is haploid, then it will have one binary vector. If it is diploid it may have two binary vectors and so on. So a het in a diploid individual would mean there is a 1 in one of the vectors at a given position, but a 0 in the other vector at the same position. This design is heavily influenced by the class of a similar name in R's adegenet. So anything we can improve on this would be really awesome!